.circleci/config.yml

```diff
@@ -0,0 +1,49 @@
+version: 2.1
+jobs:
+  test-ruby:
+    parameters:
+      ruby_version:
+        type: string
+    docker:
+      - image: "cimg/ruby:<< parameters.ruby_version >>"
+      - image: cimg/redis:6.2.6
+      - image: memcached:1.6.18
+    working_directory: ~/sdk
+    steps:
+      - checkout
+      - run:
+          name: Install dependencies
+          command: bundle install
+      - run:
+          name: Run tests
+          command: bundle exec rake
+  publish:
+    docker:
+      - image: cimg/ruby:3.2.0
+    steps:
+      - checkout
+      - run:
+          name: Install dependencies
+          command: bundle install
+      - run:
+          name: Publish to Rubygems
+          command: bundle exec rake release
+workflows:
+  version: 2
+  test-and-publish:
+    jobs:
+      - test-ruby:
+          matrix:
+            parameters:
+              ruby_version: ["3.2", "3.1", "3.0", "2.7"]
+          filters:
+            tags:
+              only: /.*/
+      - publish:
+          requires:
+            - test-ruby
+          filters:
```

Comment:
Any reason we are not using the same method as https://github.com/procore/js-sdk/blob/master/.circleci/config.yml#L62 implemented here.

Reply:
The JS SDK publishes automatically on every merge to master, but I wanted to have more control over the release. This way we can merge multiple contributor PRs and release them as as a single new version.